ruby/engine/RubyEngine.cpp

```diff
@@ -237,6 +237,25 @@ int RubyEngine::numberOfArguments(ScriptObject& methodObject, std::string_view m
   return rb_obj_method_arity(val, method_id);
 }
 
+bool RubyEngine::hasMethod(ScriptObject& methodObject, std::string_view methodName, bool overriden_only) {
+  auto val = std::any_cast<VALUE>(methodObject.object);
+  ID method_id = rb_intern(methodName.data());
+  if (rb_respond_to(val, method_id) == 0) {
+    return false;
+  }
+  if (!overriden_only) {
+    return true;
+  }
+
+  // I'd have prefered to do the equivalent of `instance_obj.method(:methodName).owner == instance_obj.class` but that is borderline impossible
+  // Instead, this is equivalent to `instance_obj.class.instance_methods(false).include?(:methodName)`
+  VALUE klass = rb_obj_class(val);
+  // include_methods_from_ancestors: false;
+  VALUE args[1] = {Qfalse};
+  VALUE methods_without_ancestors = rb_class_instance_methods(1, args, klass);
+  return rb_ary_includes(methods_without_ancestors, ID2SYM(method_id)) == Qtrue;
+}
```

Comment:
The worst was this. Ruby's C API is what it is... i found lots of methods that were better suited but none public, and it took a long time to come up with this